recipes/libdrawille/all/conanfile.py

```diff
@@ -0,0 +1,74 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.microsoft import is_msvc
+import os
+
+required_conan_version = ">=1.53.0"
+
+class LibdrawilleConan(ConanFile):
+    name = "libdrawille"
+    description = "C implementation of drawille library and extra drawing functionality"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Huulivoide/libdrawille/"
+    topics = ("drawille")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.arch not in ["x86", "x86_64"]:
+            raise ConanInvalidConfiguration(f"{self.ref} supports x86 and x86_64 only.")
+        if is_msvc(self):
+            raise ConanInvalidConfiguration(f"{self.ref} does not support MSVC.(yet)")
```

Comment:
This validate says no msvc, but the patch implies that it does? Who's right here?

Reply:
patch only says `rename restrict keyword for msvc`. it might be needed change for MSVC support, but it doesn't mean that change is sufficient on its own. it might be more patches are needed for complete MSVC support.

Reply:
That makes sense, yeah :)
It was just confusing to me that @toge added patches to something that's not build-able :)

Reply:
@RubenRBS  @SSE4 
Thank you for your comments.
Yes, I created a patch to support msvc, but I found there are other big issues:
```
C:\J\w\prod\BuildSingleReference@7\.conan\data\libdrawille\cci.20160428\_\_\build\a79a557254fabcb77849dd623fed97c9c5ab7651\src\src\utils.c(10,1): fatal error C1083: Cannot open include file: 'cpuid.h': No such file or directory [C:\J\w\prod\BuildSingleReference@7\.conan\data\libdrawille\cci.20160428\_\_\build\a79a557254fabcb77849dd623fed97c9c5ab7651\build\src\drawille.vcxproj]
```

Reply:
that's not standard [header](https://github.com/gcc-mirror/gcc/blob/master/gcc/config/i386/cpuid.h) that doesn't exist for MSVC. it requires either porting code with `#ifdef` to Microsoft `__cpuid` [instrinsic](https://learn.microsoft.com/en-us/cpp/intrinsics/cpuid-cpuidex?view=msvc-170), or providing the same header as an another package.